rust-toolchain

```diff
@@ -1 +1 @@
-nightly-2023-09-05
+1.77
```

Comment:
I will file a separate issue for upgrading to 1.78. We will need to fix some assumptions in our unsafe code first.

Reply:
I filed https://github.com/apache/datafusion-comet/issues/507

Reply:
Will this use `1.77.2` automatically? Looks like there are few fixes

Reply:
I updated this to 1.77.2